spinoso-string/src/enc/ascii/mod.rs

```diff
@@ -198,7 +198,15 @@ impl AsciiString {
     pub fn get_char_slice(&self, range: Range<usize>) -> Option<&'_ [u8]> {
         let Range { start, end } = range;
 
-        self.inner.get(start..end).or_else(|| self.inner.get(start..))
+        if start > self.len() || start > self.char_len() {
+            return None;
+        }
+
+        return match end as isize - start as isize {
+            d if d <= 0 => Some(&[]),
+            1 => self.get_char(start),
```

Comment:
I added `as isize` otherwise `end - start` would be `usize` and the `<= 0` branch would not work due to `warning: comparison is useless due to type limits`.